CommonMark/CommonMarkSettings.cs

```diff
@@ -120,12 +120,12 @@ public CommonMarkSettings Clone()
 
         #region [ Properties that cache structures used in the parsers ]
 
-        private Func<Parser.Subject, Syntax.Inline>[] _inlineParsers;
+        private Func<Parser.Subject, CommonMarkSettings, Syntax.Inline>[] _inlineParsers;
```

Comment:
would it be possible for these delegates not to take CommonMarkSettings instances? The idea of these delegates is that they are created from one specific settings object.